src/main/java/ch/njol/skript/expressions/ExprTool.java

```diff
@@ -82,7 +82,7 @@ protected Slot[] get(final Event e, final LivingEntity[] source) {
 			@Nullable
 			public Slot get(final LivingEntity ent) {
 				if (!delayed) {
-					if (e instanceof PlayerItemHeldEvent && ((PlayerItemHeldEvent) e).getPlayer() == ent) {
+					if (e instanceof PlayerItemHeldEvent && ((PlayerItemHeldEvent) e).getPlayer() == ent && !offHand) {
```

Comment:
Offhand check should come first
```suggestion
					if (!offHand && e instanceof PlayerItemHeldEvent && ((PlayerItemHeldEvent) e).getPlayer() == ent) {
```